src/data_structures/interval_tree.rs

```diff
@@ -73,6 +74,28 @@ impl<'a, N: Debug + Num + Clone + Ord + 'a, D: Debug + 'a> Iterator for Interval
     }
 }
 
+impl<N: Debug + Num + Clone + Ord, D: Debug> FromIterator<(Range<N>, D)> for IntervalTree<N, D> {
+    fn from_iter<I: IntoIterator<Item=(Range<N>, D)>>(iter: I) -> Self {
+        let mut tree = IntervalTree::new();
+
+        for r in iter {
+            tree.insert(r.0, r.1).unwrap();
+        }
+        tree
+    }                                                                                             
+}
+
+impl<'a, N: Debug + Num + Clone + Ord + 'a, D: Debug + Clone + 'a> FromIterator<Entry<'a, N, D>> for IntervalTree<N, D> {
+    fn from_iter<I: IntoIterator<Item=Entry<'a, N, D>>>(iter: I) -> Self {
+        let mut tree = IntervalTree::new();
+
+        for r in iter {
+            tree.insert(r.interval().clone(), r.data().clone()).unwrap();
```

Comment:
Still unsure if we should keep this. The implicit data cloning can be quite expensive and should be made quite clear to the user.